### PR TITLE
Simple POD class for connection info loaded from restart file

### PR DESCRIPTION
**Pull request by joakim-hove**
OK - this should be ready.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/io/eclipse/rst/connection.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    depth(scon[VI::SConn::Depth]),
+    diameter(scon[VI::SConn::Diameter]),
+    kh(scon[VI::SConn::EffectiveKH]),
+    segdist_end(scon[VI::SConn::SegDistEnd]),
+    segdist_start(scon[VI::SConn::SegDistStart]),
+    oil_rate(xcon[VI::XConn::OilRate]),
+    water_rate(xcon[VI::XConn::WaterRate]),
+    gas_rate(xcon[VI::XConn::GasRate]),
+    pressure(xcon[VI::XConn::Pressure]),
+    resv_rate(xcon[VI::XConn::Pressure])
```
Comment (bska) [inline src/opm/io/eclipse/rst/connection.cpp]: This looks a little suspicious.  Did you really mean `Pressure` here?

File: src/opm/parser/eclipse/EclipseState/Schedule/Well/WellProductionProperties.cpp [hunk 2]
```diff
@@ -352,6 +352,7 @@ namespace Opm {
         controls.alq_value = this->ALQValue;
         controls.cmode = this->controlMode;
 
+        printf("Setting controls.liquid_rate: %lg \n", controls.liquid_rate);
```
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/Schedule/Well/WellProductionProperties.cpp]: This looks like unrelated debug code.

File: tests/test_AggregateConnectionData.cpp [hunk 3]
```diff
@@ -792,6 +795,43 @@ BOOST_AUTO_TEST_CASE(InactiveCell) {
...
+                                 rptStep
+                                 );
+
+    const auto icon = conn.getIConn();
+    const auto scon = conn.getSConn();
+    const auto xcon = conn.getXConn();
+
+    std::vector<Opm::RestartIO::Connection> connections;
+    for (std::size_t iw=0; iw < header.num_wells; iw++) {
+        for (std::size_t ic=0; ic < header.ncwmax; ic++) {
```
Comment (bska) [inline tests/test_AggregateConnectionData.cpp]: This is test code so probably not a big deal.  In general, though, the well's number of active connections is IWEL[[`IWell::NConn`](https://github.com/OPM/opm-common/blob/75b64025a321072a40b75b86dc30937cc59539e9/opm/output/eclipse/VectorItems/well.hpp#L33)]

File: src/opm/parser/eclipse/EclipseState/Schedule/Well/WellProductionProperties.cpp (hunk 2 above)
Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Schedule/Well/WellProductionProperties.cpp]: Removed

File: src/opm/io/eclipse/rst/connection.cpp (hunk 1 above)
Comment (joakim-hove) [reply src/opm/io/eclipse/rst/connection.cpp]: Ehhh - no - fixed

File: tests/test_AggregateConnectionData.cpp (hunk 3 above)
Comment (joakim-hove) [reply tests/test_AggregateConnectionData.cpp]: Yes I know; the test is written "before" IWEL is created.

**Comment by joakim-hove** [comment]
> I still would have preferred that the Connection constructor be defined in terms of boost::iterator_range<std::vector<T>::const_iterator> rather than const T* pointers, but this is at least better than vector<T> parameter types.

I must admit I am reluctant to introduce this boost construction; `const T* works` and "everyone" understands it. I am all for utilizing new functionality from the `std::` namespace when it comes along, but more skeptical to take detours via boost. In this particular case I would say that `const T*` works and everybody understands it, `boost::iterator_range< >` will impose a cognitive burden on future readers (and me right now - I am a lazy bastard). So - all in all; I would prefer to leave the `const T*` approach intact.

**Comment by joakim-hove** [comment]
jenkins build this please
